wpimath/src/test/java/edu/wpi/first/math/optimization/SimulatedAnnealingTest.java

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package edu.wpi.first.math.optimization;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.VecBuilder;
+import edu.wpi.first.math.Vector;
+import edu.wpi.first.math.numbers.N1;
+import java.util.function.DoubleFunction;
+import org.junit.jupiter.api.Test;
+
+class SimulatedAnnealingTest {
+  @Test
+  void testDoubleFunctionOptimizationHeartBeat() {
+    DoubleFunction<Double> function = (x) -> -(x + Math.sin(x)) * Math.exp(-x * x) + 1;
+
+    double stepSize = 10.0;
+
+    Vector<N1> ans =
+        SimulatedAnnealing.minimize(
+            VecBuilder.fill(-1),
+            5000,
+            2.0,
```

Comment:
Are these sane defaults the user doesn't have to specify manually?

Reply:
I think these should probably be in the hands of the user.
For the initial state, the user might have a good estimation that they want to start off + they can tune how many iterations are required for their own use case (same goes for stepsize).

Reply:
Then how to properly select these values needs to be documented.